### PR TITLE
add vue-material@0.7.1 w/ npm auto-update

### LINKED ISSUE #9992: [Request] Add Vue Material
**Library name:** Vue Material
**Git repository url:** https://github.com/marcosmoura/vue-material
**npm package url(optional):** https://www.npmjs.com/package/vue-material
**License(s):** MIT
**Official homepage:** https://vuematerial.github.io

### PR DESCRIPTION
**Pull request by sufuf3**
close #9992, cc @marcosmoura
@kennynaoh please help me review this PR, thank you.

Pull request for issue: #9992 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/marcosmoura/vue-material
 * Official website (optional, not the repository): https://vuematerial.github.io
 * NPM package url (optional): https://www.npmjs.com/package/vue-material
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 97
   - Count of stars: 1708
   - Count of forks: 170
 * NPM download stats (optional):
   - Downloads in the last day: 111
   - Downloads in the last week: 1451
   - Downloads in the last month: 5651

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/vue-material/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "angular-material",
+    "components"
+  ],
+  "filename": "vue-material.js",
+  "npmName": "vue-material",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/vue-material.+(css*|js)"
```
Comment (PeterDaveHello) [inline ajax/libs/vue-material/package.json]: Why `css*` here?

Comment (PeterDaveHello) [inline ajax/libs/vue-material/package.json]: Don't we need `components`?

**Comment by sufuf3** [comment]
I modified it.
@kennynaoh please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
You should also provide the license reference in comments next time.

File: ajax/libs/vue-material/package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "components"
+  ],
+  "filename": "vue-material.js",
+  "npmName": "vue-material",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/vue-material.!(debug)*",
+        "components/*/*"
```
Comment (maruilian11) [inline ajax/libs/vue-material/package.json]: how about `index.debug.js` in `components`?

File: ajax/libs/vue-material/package.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    "vue",
+    "vue 2",
+    "vue2",
+    "vuejs",
+    "material design",
+    "material",
+    "angular-material",
+    "components"
+  ],
+  "filename": "vue-material.js",
```
Comment (maruilian11) [inline ajax/libs/vue-material/package.json]: put it in previous field

**Comment by maruilian11** [comment]
why do you need to use `dist/vue-material.js` as basepath?

**Comment by sufuf3** [comment]
I modified it.
@maruilian11  please help me review this PR again, thank you.

**Comment by PeterDaveHello** [comment]
The map file will not hurt.

**Comment by PeterDaveHello** [comment]
ping @kennynaoh

**Comment by kennynaoh** [comment]
@sufuf3 Sorry for late response. I think this PR is ok. Please rebase to the latest master branch. Thanks!

File: ajax/libs/vue-material/package.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    "angular-material",
+    "components"
+  ],
+  "npmName": "vue-material",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "vue-material!(*debug*)",
+        "components/*/!(*debug*)"
```
Comment (pvnr0082t) [inline ajax/libs/vue-material/package.json]: The files with `debug` are the non-minified files. We can include them.

Comment (maruilian11) [inline ajax/libs/vue-material/package.json]: @PeterDaveHello do we also host non-minified debug `js` filed?

Comment (PeterDaveHello) [inline ajax/libs/vue-material/package.json]: Looks fine in this case.

**Comment by sufuf3** [comment]
I modified it.
@pvnr0082t  please help me review this PR again, thank you.

File: ajax/libs/vue-material/package.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "angular-material",
+    "components"
+  ],
+  "npmName": "vue-material",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/vue-material*",
+        "components/**/*"
```
Comment (kennynaoh) [inline ajax/libs/vue-material/package.json]: @sufuf3 The `docs/` will also be grab from upstream.

Comment (maruilian11) [inline ajax/libs/vue-material/package.json]: please test the auto-update config before you send the PR.

**Comment by sufuf3** [comment]
I modified it.
@pvnr0082t please help me review this PR again, thank you.

**Comment by PeterDaveHello** [comment]
hmmmm ... we don't need to minify those `vue-material.debug.js`

**Comment by maruilian11** [comment]
@PeterDaveHello do our bot support partly minify files function? or we need to add another auto-update filemap for those `debug.js`s?

**Comment by PeterDaveHello** [comment]
>  do our bot support partly minify files function?

No, `do` -> `does` ...

**Comment by maruilian11** [comment]
@PeterDaveHello so we choose to not minify all files in this case?

**Comment by pvnr0082t** [comment]
@maruilian11 Yes. Because it seems `vue-material.debug.js` is the non-minified file of `vue-material.js`, we don't need to minify it. 
Just the naming problem. In common naming scheme, `vue-material.debug.js` is just like `*.js` and `vue-material.js` is like `*.min.js`. :+1:

**Comment by sufuf3** [comment]
@PeterDaveHello  I remove the `*.debug.min.js*` files.

**Comment by kennynaoh** [comment]
LGTM

File: ajax/libs/vue-material/package.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    "angular-material",
+    "components"
+  ],
+  "npmName": "vue-material",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "vue-material*",
+        "components/**/*"
```
Comment (PeterDaveHello) [inline ajax/libs/vue-material/package.json]: Should use blacklist in `dist` not whitelist ...
Try `"!(docs)/**/*"`

**Comment by sufuf3** [comment]
@PeterDaveHello I modified it.
please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
LGTM

**Pull request merged**